pennylane/pulse/hardware_hamiltonian.py

```diff
@@ -194,17 +196,20 @@ def circuit_local(params):
 
             H_d = qml.pulse.drive(amplitude, phase, wires)
```

Comment:
Just a note for a couple of things:
* I'm a little confused about once again. In my previous PR, the detuning no longer had a coefficient of 1/2, but the equation shown above (line 187) seems to suggest otherwise. It doesn't make much of a difference right now, but it might still be worthwhile being more consistent about it.
* I can't add this comment to the line myself for some reason, but the definition of `wires = [1, 2, 3, 4]` on line 179 should be `wires = [0, 1, 2, 3]`.
* It might make a bit more sense to put the Hamiltonian expression on line 187 **after** the code block where the amplitude, phase and detuning are defined.

Reply:
Re factor 1/2, I think we just need to **decide on one convention and stick with that**. From your PR I take it you follow the bloqade convention (which makes most sense since we are targeting aquila device), i.e. 1/2 in front of drive, no 1/2 in front of detuning. I change it to that, good call!


For neutral atoms it seems to be common to use explicit 1/2 factors, but even then they are not consistent where they put it:

[pulser](https://pulser.readthedocs.io/en/stable/review.html) has it for both
![image](https://user-images.githubusercontent.com/43949391/230331875-521797a3-7872-4654-abd8-708f518b133f.png)

[bloqade](https://queracomputing.github.io/Bloqade.jl/dev/hamiltonians/) has it only on the drive term
![image](https://user-images.githubusercontent.com/43949391/230332052-05a9aaec-46f8-47b1-918b-1c542e3adc19.png)

In the [ctrl-vqe paper](https://arxiv.org/pdf/2203.06818.pdf) they dont put explicit 1/2 factors
![image](https://user-images.githubusercontent.com/43949391/230332342-831686ad-5f75-49ef-8551-ac93875e4ed0.png)

Reply:
What is the problem with having wires `[1, 2, 3, 4]`?
![image](https://user-images.githubusercontent.com/43949391/230333445-4619e5b7-4588-4599-abc1-0c5552c24d1f.png)

Reply:
> It might make a bit more sense to put the Hamiltonian expression on line 187 after the code block where the amplitude, phase and detuning are defined.

Agreed, done :)

Reply:
> What is the problem with having wires [1, 2, 3, 4]?

The wires themselves aren't problematic. I thought that the device wires were just initialized as `dev = qml.device("default.qubit.jax", wires=4)` rather than `dev = qml.device("default.qubit.jax", wires=wires)`. This is fine I just didn't read the device initialization properly 😅